packages/starlight/utils/routing.ts

```diff
@@ -100,6 +100,22 @@ function getRoutes(): Route[] {
 }
 export const routes = getRoutes();
 
+function getParamRouteMapping(): ReadonlyMap<string | typeof INDEX_SLUG_PARAM, Route> {
+	const map = new Map<string | typeof INDEX_SLUG_PARAM, Route>();
+
+	for (const route of routes) {
+		map.set(slugToParam(route.slug) ?? INDEX_SLUG_PARAM, route);
+	}
+
+	return map;
+}
+const INDEX_SLUG_PARAM = Symbol('index');
+const routesBySlugParam = getParamRouteMapping();
+
+export function getRouteBySlugParam(slugParam: string | undefined): Route | undefined {
+	return routesBySlugParam.get(slugParam?.replace(/\/$/, '') ?? INDEX_SLUG_PARAM);
+}
```

Comment:
Couple of questions here:
- is there a particular advantage of using a `Symbol` vs just using `undefined` as the index key?
- should the `??` be `||`? I’d guess a param of `""` should be handled the same as a param of `undefined`

```suggestion
function getParamRouteMapping(): ReadonlyMap<string | undefined, Route> {
	const map = new Map<string | undefined, Route>();
	for (const route of routes) {
		map.set(slugToParam(route.slug), route);
	}
	return map;
}
const routesBySlugParam = getParamRouteMapping();

export function getRouteBySlugParam(slugParam: string | undefined): Route | undefined {
	return routesBySlugParam.get(slugParam?.replace(/\/$/, '') || undefined);
}
```

Reply:
> is there a particular advantage of using a `Symbol` vs just using `undefined` as the index key?

Matter of preference, I dislike using `undefined` as a marker for things, but it works the same.

> should the `??` be `||`? I’d guess a param of `""` should be handled the same as a param of `undefined`

Both `slugToParam` and the Astro params from the URL transform the root index to `undefined` so there is no case for the empty string. So it could be `||` with no real difference.